src/target/puya.c

```diff
@@ -0,0 +1,171 @@
+#include "general.h"
```

Comment:
This file must have a license header included and it must go before the first `#include` please. We cannot take the contribution otherwise. Either BSD-3-Clause or GPLv3+ is fine (our preference is BSD-3-Clause, but this is up to you).